### PR TITLE
feat: have wiki with MyAEGEE auth

### PR DESCRIPTION
**Pull request by linuxbandit**
I want to add a maria provisioning so we can already mount the localsettings (assuming it has been filled with the right info)

after that and your approval, i will merge

### CODE REVIEW

File: oms-wiki/docker/TODO [hunk 1]
```diff
@@ -1 +1,3 @@
 generate LocalSettings.php from the values in the env file
```
Comment (serge1peshcoff) [inline oms-wiki/docker/TODO]: Why not substituting $_ENV["something"] instead? We won't need the envsubst and the example file there that way

File: oms-wiki/docker/LocalSettings.php.example [hunk 2]
```diff
@@ -30,7 +30,7 @@ $wgMetaNamespace = "AEGEE_wiki";
 $wgScriptPath = "";
 
 ## The protocol and server name to use in fully-qualified URLs
-$wgServer = "https://wiki.staging.barnab.eu";
+$wgServer = "https://wiki.${BASE_URL}";
```
Comment (serge1peshcoff) [inline oms-wiki/docker/LocalSettings.php.example]: `$wgServer = "https://".$_ENV["SUBDOMAIN_WIKI"].$_ENV["BASE_URL"]`

Comment (linuxbandit) [reply oms-wiki/docker/LocalSettings.php.example]: ooh that would be brilliant! good point, i forgot it's php and not just a configuration language!

Comment (serge1peshcoff) [reply oms-wiki/docker/LocalSettings.php.example]: Also the wiki tokens should most likely to to the global .env and passed to this config the same way so we'll be secure

File: oms-wiki/docker/docker-compose.yml [hunk 3]
```diff
@@ -11,25 +11,23 @@ services:
...
   mediawiki:
     image: mediawiki:1.33
     restart: always
-    expose:
-        - "80"
     volumes:
       - mediawiki-images:/var/www/html/images
+      - ./${PATH_OMS_WIKI}/MyAEGEEAuth:/var/www/html/extensions/MyAEGEEAuth
       # After initial setup, download LocalSettings.php to the same directory as
       # this yaml and uncomment the following line and use compose to restart
```
Comment (serge1peshcoff) [inline oms-wiki/docker/docker-compose.yml]: We can uncomment this one, I suppose the only things changing there in LocalSettings.php are secrets, and given that we'll move them to global .env, we're okay

File: oms-wiki/docker/LocalSettings.php.example [hunk 4]
```diff
@@ -30,33 +30,33 @@ $wgMetaNamespace = "AEGEE_wiki";
 $wgScriptPath = "";
 
 ## The protocol and server name to use in fully-qualified URLs
-$wgServer = "https://wiki.staging.barnab.eu";
+$wgServer = "https://".$_ENV["SUBDOMAIN"].$_ENV["URL"];
```
Comment (serge1peshcoff) [inline oms-wiki/docker/LocalSettings.php.example]: Why not BASE_URL?

File: oms-wiki/docker/docker-compose.yml (hunk 3 above)
Comment (serge1peshcoff) [reply oms-wiki/docker/docker-compose.yml]: Also let's delete this comment at all, it's not relevant anymore

File: oms-wiki/docker/LocalSettings.php.example (hunk 4 above)
Comment (linuxbandit) [reply oms-wiki/docker/LocalSettings.php.example]: i changed the var names inside

Comment (serge1peshcoff) [reply oms-wiki/docker/LocalSettings.php.example]: reasonable. I'd rather leave it as SUBDOMAIN_WIKI and BASE_URL, but that's not that important, it'll work that way as well.

File: oms-wiki/docker/docker-compose.yml (hunk 3 above)
Comment (linuxbandit) [reply oms-wiki/docker/docker-compose.yml]: agree on deleting the comment, what are you saying to uncomment?

Comment (serge1peshcoff) [reply oms-wiki/docker/docker-compose.yml]: the LocalSettings.php mounting, I see you've updated it as well already, so it's not relevant anymore

File: oms-wiki/docker/LocalSettings.php.example (hunk 4 above)
Comment (linuxbandit) [reply oms-wiki/docker/LocalSettings.php.example]: fine

**Comment by serge1peshcoff** [comment]
one thing, can you move `$wgSecretKey` and `$wgUpgradeKey` from there to global `.env` and set it to something simple there?

**Comment by linuxbandit** [comment]
will do right after this check of user account creation

File: oms-wiki/docker/MyAEGEEAuth/includes/MyAEGEEApiPrimaryAuthenticationProvider.php [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+		}
+
+		return $response_parsed->data;
+	}
+
+	/**
+	 * All fun starts here.
+	 */
+	public function beginPrimaryAuthentication( array $reqs ) {
+		wfDebugLog( 'MyAEGEEApi', json_encode($reqs[0]) );
```
Comment (serge1peshcoff) [inline oms-wiki/docker/MyAEGEEAuth/includes/MyAEGEEApiPrimaryAuthenticationProvider.php:156]: forgot to remove this one, can you?

Comment (linuxbandit) [reply oms-wiki/docker/MyAEGEEAuth/includes/MyAEGEEApiPrimaryAuthenticationProvider.php:156]: as per slack, you'll do in the next PR

**Review by serge1peshcoff — APPROVED** [review]

**Pull request merged**